test/razorTests/razorTelemetry.integration.test.ts

```diff
@@ -0,0 +1,55 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import testAssetWorkspace from '../integrationTests/testAssets/testAssetWorkspace';
+import { activateCSharpExtension } from '../integrationTests/integrationHelpers';
+import { htmlLanguageFeaturesExtensionReady } from "./testUtils";
+import { EventStream } from '../../src/EventStream';
+import { EventType } from '../../src/omnisharp/EventType';
+import { TelemetryEvent } from 'microsoft.aspnetcore.razor.vscode/dist/HostEventStream';
+import { Subscription } from 'rxjs';
+
+let eventStream: EventStream;
+let subscription: Subscription;
+
+suite(`Razor Telemetry ${testAssetWorkspace.description}`, () => {
+    suiteSetup(async function () {
+        eventStream = (await activateCSharpExtension()).eventStream;
+        await htmlLanguageFeaturesExtensionReady();
+        await testAssetWorkspace.restore();
+    });
+
+    suiteTeardown(async () => {
+        await testAssetWorkspace.cleanupWorkspace();    
+    });
+
+    teardown(() => {
+        if (subscription) {
+            subscription.unsubscribe();
+       } 
+    });
+
+    test("The DocumentOpened telemevent is received when a document is opened", async () => {
+        let telemetryEventResolver: (value?: any) => void;
+        // tslint:disable-next-line:promise-must-complete
+        const extensionActivated = new Promise(resolve => {
+            telemetryEventResolver = resolve;
+        });
+        
+        subscription = eventStream.subscribe(event =>{
+            if (event.type == EventType.TelemetryEvent) {
+                let possibleRazorEvent = <TelemetryEvent>event;
+                if (possibleRazorEvent.eventName == "VSCode.Razor.DocumentOpened") {
```

Comment:
👏  Would probably be good to also ensure that you don't receive: `VSCode.Razor.ErrorOnActivation`

Reply:
I am not sure if we can make it work. By the time we obtain the event stream and subscribe to it,  the "ErrorOnActivation" would have already been raised and I dont think there is a way currently to get the past events from the event stream.

Reply:
Thoughts ?

Reply:
Ah, if it's out of reach then don't worry about it.